zhaquirks/tuya/ts0601_trv_maxsmart.py

```diff
@@ -0,0 +1,1294 @@
+MAXSMART_AWAY_DATA_ATTR = 0x0067
+MaxsmartManufClusterSelf = None
+SILVERCREST_BATTERY_ATTR = 0x0223  # battery charge
+SILVERCREST_CHILD_LOCK_ATTR = 0x0128  # [0] unlocked [1] child-locked
+
+
+class MaxsmartManufCluster(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of some thermostatic valves."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.MaxsmartManufCluster_bus.add_listener(self)
+        global MaxsmartManufClusterSelf
+        MaxsmartManufClusterSelf = self
+
+    set_time_offset = 1970
+
+    manufacturer_attributes = {
+        MAXSMART_CHILD_LOCK_ATTR: ("child_lock", t.uint8_t),
+        MAXSMART_WINDOW_DETECT_ATTR: ("window_detection", t.uint8_t),
+        MAXSMART_WINDOW_DETECT_TEMP_ATTR: ("window_detection_temp", t.uint32_t),
+        MAXSMART_WINDOW_DETECT_TIME_ATTR: ("window_detection_time", t.uint32_t),
+        MAXSMART_TARGET_TEMP_AUTO_ATTR: ("target_temperature_auto", t.uint32_t),
+        MAXSMART_TARGET_TEMP_MAN_ATTR: ("target_temperature_manual", t.uint32_t),
+        MAXSMART_TEMPERATURE_ATTR: ("temperature", t.uint32_t),
+        MAXSMART_MODE_ATTR: ("mode", t.uint8_t),
+        MAXSMART_TEMP_CALIBRATION_ATTR: ("temperature_calibration", t.int32s),
+        MAXSMART_COMFORT_TEMP_ATTR: ("comfort_mode_temperature", t.uint32_t),
+        MAXSMART_ECO_TEMP_ATTR: ("eco_mode_temperature", t.uint32_t),
+        MAXSMART_BATTERY_ATTR: ("battery", t.uint32_t),
+        MAXSMART_AWAY_DATA_ATTR: ("away_data", t.data64),
+    }
+
+    DIRECT_MAPPED_ATTRS = {
+        MAXSMART_TEMPERATURE_ATTR: ("local_temp", lambda value: value * 10),
+        MAXSMART_TARGET_TEMP_AUTO_ATTR: (
+            "occupied_heating_setpoint",
+            lambda value: value / 2 * 100,
+        ),
+        MAXSMART_TARGET_TEMP_MAN_ATTR: (
+            "occupied_heating_setpoint",
+            lambda value: value / 2 * 100,
+        ),
+        MAXSMART_AWAY_DATA_ATTR: (
+            "unoccupied_heating_setpoint",
+            lambda value: value[2] / 2 * 100,
+        ),
+        MAXSMART_COMFORT_TEMP_ATTR: (
+            "comfort_heating_setpoint",
+            lambda value: value / 2,
+        ),
+        MAXSMART_ECO_TEMP_ATTR: ("eco_heating_setpoint", lambda value: value / 2),
+        MAXSMART_TEMP_CALIBRATION_ATTR: (
+            "local_temperature_calibration",
+            lambda value: value * 10,
+        ),
+        MAXSMART_WINDOW_DETECT_TEMP_ATTR: (
+            "window_detection_temp",
+            lambda value: value / 2,
+        ),
+        MAXSMART_WINDOW_DETECT_TIME_ATTR: ("window_detection_time", None),
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+
+        if attrid in self.DIRECT_MAPPED_ATTRS:
+            self.endpoint.device.thermostat_bus.listener_event(
```

Comment:
We can stop using bus and we can do something like this: https://github.com/zigpy/zha-device-handlers/blob/10dbcc4d64fdf748989e6d72ea169bc7163eb6ad/zhaquirks/xiaomi/aqara/tvoc.py#L31 I would like to eventually remove Bus from the codebase altogether

Reply:
So this example is calling using ep_attribute name directly??

Reply:
> We can stop using bus and we can do something like this:
> 
> https://github.com/zigpy/zha-device-handlers/blob/10dbcc4d64fdf748989e6d72ea169bc7163eb6ad/zhaquirks/xiaomi/aqara/tvoc.py#L31
> I would like to eventually remove Bus from the codebase altogether

The `self.endpoint.device.thermostat_bus` comes from inherited class `TuyaThermostatCluster`.

https://github.com/MattWestb/zha-device-handlers/blob/01f7ed0787319c2da85c05fa242c829f334b0cb4/zhaquirks/tuya/__init__.py#L513

At first we have to remove the bus from this class and then rewrite all tuya TRV quirks. Which cannot be tested, because nobody owns all TRVs. Or we could create new class `TuyaThermostatCluster2` without bus and slowly migrate all the TRVs.